### PR TITLE
added markdown preview

### PR DESCRIPTION
**Pull request by SashaBondarchuk**
https://trello.com/c/DfHaAokq/209-bug83-main-page-the-challenge-description-is-absent-in-the-suggested-challenge-block
https://trello.com/c/I6VExyBM/226-bug92-challenge-page-when-the-challenge-page-opens-the-header-is-half-visible

image

image

### CODE REVIEW

**Comment by SashaBondarchuk** [comment]
[[BUG#92] [Challenge page] When the challenge page opens, the header is half visible](https://trello.com/c/I6VExyBM/226-bug92-challenge-page-when-the-challenge-page-opens-the-header-is-half-visible)

**Review by bodichka11 — APPROVED** [review]

**Comment by SashaBondarchuk** [comment]
[[BUG#110] [Editor challenge page] The page does not open correctly](https://trello.com/c/KbTgocEa/251-bug110-editor-challenge-page-the-page-does-not-open-correctly)

**Comment by SashaBondarchuk** [comment]
[[BUG#99] [Profile page] The main info block changes the size according to the username](https://trello.com/c/HtzPFYcu/236-bug99-profile-page-the-main-info-block-changes-the-size-according-to-the-username)

**Comment by SashaBondarchuk** [comment]
https://trello.com/c/KbTgocEa/251-bug110-editor-challenge-page-the-page-does-not-open-correctly

https://trello.com/c/HtzPFYcu/236-bug99-profile-page-the-main-info-block-changes-the-size-according-to-the-username

File: frontend/src/app/modules/challenges/challenge-creation/challenge-creation.component.ts [hunk 1]
```diff
@@ -90,6 +90,7 @@ export class ChallengeCreationComponent extends BaseComponent implements HasUnsa
     }
 
     ngOnInit(): void {
+        window.scrollTo({ top: 0, behavior: 'smooth' });
```
Comment (tatianahutii) [inline frontend/src/app/modules/challenges/challenge-creation/challenge-creation.component.ts]: I believe this is more of a workaround than fixing the core root cause.

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by SashaBondarchuk** [comment]
[[BUG#89] [Auth] Refresh token functionality does not work](https://trello.com/c/wZ9cayDg/215-bug89-auth-refresh-token-functionality-does-not-work)

**Comment by SashaBondarchuk** [comment]
image

File: frontend/src/app/core/services/auth.service.ts [hunk 2]
```diff
@@ -48,11 +48,22 @@ export class AuthService {
...
+    refreshFirebaseToken(): Observable<string> {
+        return this.afAuth.authState.pipe(
+            switchMap((user) => {
+                if (user) {
+                    return from(user.getIdToken(true));
+                }
+                throw new Error('User not authenticated');
+            }),
+            catchError((error) => {
+                console.error('Error refreshing Firebase token:', error);
```
Comment (tatianahutii) [inline frontend/src/app/core/services/auth.service.ts]: please delete all console.error

File: frontend/src/app/shared/components/header/header.component.ts [hunk 3]
```diff
@@ -153,6 +142,43 @@ export class HeaderComponent extends BaseComponent implements OnInit, OnDestroy
...
+        this.notificationStorage
+            .getUserNotifications()
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next: (notifications) => {
+                    this.newNotificationsCollection = notifications.filter((e) => !e.isRead);
+                    this.seenNotificationsCollection = notifications.filter((e) => e.isRead);
+                },
+                error: () => {
+                    this.toastrService.showError('Server connection error');
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/header/header.component.ts]: please show error message from our backend

File: frontend/src/app/shared/components/header/header.component.ts [hunk 4]
```diff
@@ -153,6 +142,43 @@ export class HeaderComponent extends BaseComponent implements OnInit, OnDestroy
...
+        this.newNotificationsCollection = [];
+    }
+
+    private readNotificationsAndUpdateStatus() {
+        this.notificationStorage
+            .updateStatusToRead([this.user.id])
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                error: () => {
+                    this.toastrService.showError('Server connection error');
```
Comment (tatianahutii) [inline frontend/src/app/shared/components/header/header.component.ts]: the same

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by SashaBondarchuk** [comment]
[[BUG#125] [Profile page] When the user has a long name, the profile page will not adapt](https://trello.com/c/3zaTpe0d/277-bug125-profile-page-when-the-user-has-a-long-name-the-profile-page-will-not-adapt)

**Review by tatianahutii — APPROVED** [review]

**Comment by SashaBondarchuk** [comment]
[[BUG#121] [Notifications]The notification's menu doesn't close and counter doesn't disappear after clicking on the notification](https://trello.com/c/2Emdf2F0/258-bug121-notificationsthe-notifications-menu-doesnt-close-and-counter-doesnt-disappear-after-clicking-on-the-notification)

**Pull request merged**